base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
```

Comment:
- [ ] Kindly make sure the nameable variable is accessible, you can do this by adding `attr_accessor :nameable` to your class